src/ui/reportView/callReport.ts

```diff
@@ -95,22 +104,74 @@ function createCommand(commandURI: string, harnessFile: string, harnessName: str
 	return { finalCommand, searchDir };
 }
 
-// 	Find the path of the report from the harness name
-function findPath(dir: string, filename: string): string {
-	const files: string[] = fs.readdirSync(dir);
+/**
+ * Run the visualize command to generate the report, parse the output to return the python server command and status
+ *
+ * @param command - the cargo kani | kani command to run --visualize
+ * @returns - A promise of the python command and the status code; Promise<visualizeOutput>
+ */
+async function runVisualizeCommand(command: string, harnessName: string): Promise<visualizeOutput> {
+	try {
+		vscode.window.showInformationMessage(`Generating viewer report for ${harnessName}`);
+		const { stdout, stderr } = await execPromise(command);
+		const serveReportCommand: string = await parseReportOutput(stdout);
```

Comment:
nit - i think you meant `serverReportCommand` instead of `serveReportCommand` — not worth another revision